spec/services/filesync/filesync.spec.ts

```diff
@@ -1653,22 +1654,18 @@ describe("FileSync.sync", () => {
 });
 
 describe("FileSync.push", () => {
-  beforeEach(() => {
-    loginTestUser();
-    nockTestApps();
-  });
-
-  it("automatically sends local changes to gadget when gadget hasn't made any changes", async () => {
-    const { ctx, filesync, expectDirs, expectLocalAndGadgetHashesMatch } = await makeSyncScenario({
-      ctx: makeContext({ parse: PushArgs, argv: ["push", `--app=${testApp.slug}`, `--env=${testApp.environments[0]!.name}`] }),
-      localFiles: {
-        "local-file.js": "// local",
-      },
-    });
+  describeWithAuth(() => {
```

Comment:
I stole a whole bunch of code from vitest's repo to recreate a describable suite with chaining. So for example:

`describeWithAuth` will run with both token and cookies authentication
where as `describeWithAuth.token` will only run with token.

Because I'm using the same implementation as vitest it wouldn't be hard to implement `skip` `skipIf` and all the other helpers that come along with the other vite functions.

Reply:
Let me know how you feel about that @scott-rc